### PR TITLE
Feature/axios signup sign in

### PR DESCRIPTION
**Pull request by yavetisyan**

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,40 +1,60 @@
...
+      }
+    })()
+  }, [])
 
   return (
     <EventContextProvider>
-      <Box className="bg">{userId ? <PrivateRouts /> : <PublicRoutes />}</Box>
+      {!isLoading ? (
+        <Box className="bg">
+          {localStorage.token || isAuth ? <PrivateRouts /> : <PublicRoutes />}
```
Comment (ArthurO2002) [inline src/App.tsx:49]: There can be bug if token is wrong

File: src/axiosInstance.tsx [hunk 2]
```diff
@@ -1,7 +1,7 @@
 import axios from 'axios'
 
 const axiosInstance = axios.create({
-  baseURL: '',
+  baseURL: 'http://161.35.144.15:3000/',
```
Comment (ArthurO2002) [inline src/axiosInstance.tsx:4]: keep in env

File: src/pages/auth/SignIn.tsx [hunk 3]
```diff
@@ -15,31 +14,57 @@ import {
...
+            firstName: user.firstName,
+            lastName: user.lastName,
+            authUid: user.authUid,
+            email: user.email,
+            salary: user.salary,
+            userId: user.id,
+          }),
+        )
+      } catch (err) {
+        console.log(err)
```
Comment (ArthurO2002) [inline src/pages/auth/SignIn.tsx:56]: With above errors can we add alerts for errors?

File: src/redux/user/userSlice.ts [hunk 4]
```diff
@@ -2,29 +2,43 @@ import {createSlice} from '@reduxjs/toolkit'
...
   user: '',
   email: '',
-  uid: '',
+  userId: 0,
+  authUid: '',
+  salary: 0,
+  isAuth: false,
   token: '',
-  isAuth: true,
+  avatar: 'http://res.cloudinary.com/avoo/image/upload/v1652271069/v35k9wpyszjbxhyl8xn0.jpg',
```
Comment (ArthurO2002) [inline src/redux/user/userSlice.ts:14]: Why is FE setting default image for user????
It should be done from BE image

File: src/redux/user/userSlice.ts [hunk 5]
```diff
@@ -2,29 +2,43 @@ import {createSlice} from '@reduxjs/toolkit'
...
   reducers: {
     setUser(state, action) {
+      state.firstName = action.payload.firstName
+      state.lastName = action.payload.lastName
       state.user = action.payload.user
       state.email = action.payload.email
-      state.uid = action.payload.uid
+      state.authUid = action.payload.authUid
+      state.userId = action.payload.userId
+      state.isAuth = true
```
Comment (ArthurO2002) [inline src/redux/user/userSlice.ts:28]: Why always true

**Review by ArthurO2002 — CHANGES_REQUESTED** [review]

File: src/axiosInstance.tsx (hunk 2 above)
Comment (yavetisyan) [reply src/axiosInstance.tsx:4]: Doe

File: src/redux/user/userSlice.ts (hunk 4 above)
Comment (yavetisyan) [reply src/redux/user/userSlice.ts:14]: Done

File: src/redux/user/userSlice.ts (hunk 5 above)
Comment (yavetisyan) [reply src/redux/user/userSlice.ts:28]: When you sign in or sign up, in redux it goes to true, for loading, or we can give loading if we have userId

File: src/axiosInstance.tsx [hunk 6]
```diff
@@ -1,7 +1,7 @@
 import axios from 'axios'
 
 const axiosInstance = axios.create({
-  baseURL: '',
+  baseURL: process.env.REACT_APP_AXIOS_AUTH_URL,
```
Comment (ArthurO2002) [inline src/axiosInstance.tsx:4]: Chnage env name to be service name, ask @Zt0  or @avo2003

File: src/pages/auth/SignIn.tsx [hunk 7]
```diff
@@ -15,31 +14,57 @@ import {
...
-      if (auth.currentUser) {
+      setIsAuthLoading(true)
+      const auth = await axiosInstance.post('/users/signIn', {
+        email,
+        password,
+      })
+      localStorage.setItem('token', auth.data)
+
+      const user = await afterSelf(auth.data)
+      console.log(user)
```
Comment (ArthurO2002) [inline src/pages/auth/SignIn.tsx:47]: remove

File: src/pages/auth/SignUp.tsx [hunk 8]
```diff
@@ -14,24 +14,31 @@ import {
...
   const [email, setEmail] = useState<string>('')
   const [password, setPassword] = useState<string>('')
   const [nameError, setNameError] = useState<boolean>(false)
   const [emailError, setEmailError] = useState<boolean>(false)
   const [paswordError, setPasswordError] = useState<boolean>(false)
 
+  const regName = name.trim().split(' ')
+  const firstName = regName[0]
+  const lastName = regName[1]
+
```
Comment (ArthurO2002) [inline src/pages/auth/SignUp.tsx:41]: Wrong logic

**Review by ArthurO2002 — CHANGES_REQUESTED** [review]

**Pull request merged**